src/config.rs

```diff
@@ -416,4 +464,39 @@ mod tests {
         let config = HpkeClientConfig { public_key };
         assert_eq!(format!("{config:?}"), "HpkeClientConfig { public_key: \"2bd9da78f01d8bc6948bbcbe44ec1e7163d05083e267d110cdb2e75d847e3b6f\" }");
     }
+
+    #[test]
+    fn client_config_serde() {
+        fn assert_config_eq(config_str: &str, expected: &ClientConfig) {
+            let actual: ClientConfig = serde_json::from_str(config_str).unwrap();
+
+            match (&expected.http_config, &actual.http_config) {
+                (HttpClientConfigurator::Http2(left), HttpClientConfigurator::Http2(right)) => {
+                    assert_eq!(left, right);
+                }
+                (HttpClientConfigurator::Http1(_), HttpClientConfigurator::Http1(_)) => {}
+                _ => panic!(
+                    "http config is not the same: {:?} vs {:?}",
+                    expected.http_config, actual.http_config
+                ),
+            };
+        }
+
+        assert_config_eq(
+            r#"{ "http_config": { "version": "http2" } }"#,
+            &ClientConfig::configure_http2(Http2Configurator {
+                ping_interval: None,
+            }),
+        );
+        assert_config_eq(
+            r#"{ "http_config": { "version": "http1" } }"#,
+            &ClientConfig::use_http1(),
+        );
```

Comment:
And if you get `{ "version": "http1", "ping_interval_secs": "100" }`, what then?

Reply:
we get runtime error. I added a test to validate this behaviour